tests/testthat/test-test_dir.R

```diff
@@ -45,3 +45,23 @@ test_that('test_dir() helpers', {
   df <- as.data.frame(res)
   expect_true(all(!df$error & df$failed == 0))
 })
+
+test_that('find_test_scripts() with tricky names', {
```

Comment:
Rather than mocking, I'd rather that you extract out the pure code from `find_test_scripts()` - maybe call it `filter_test_scripts()`?